docs/DeveloperGuide.md

```diff
@@ -1,34 +1,383 @@
+- [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+- [User stories](#user-stories)
+- [Use cases](#use-cases)
+- [Non-functional requirements](#non-functional-requirements) 
+- [Glossary](#glossary)
+- [Instruction for manual testing](#instructions-for-manual-testing)
+
+
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+
+
+![Architecture](diagrams/Architecture.png)
+
+
+The Architecture Diagram above gives an overview of how different components interacts 
+with each other in the app. 
+
+As the app starts, the main program initialises the Ui, Logic and Storage components.
+Then, the Storage component reads existing data file if it is available, else a fresh copy of 
+data file is created. This step also initialises the content of runtime storage which is store 
+in a TaskMap object.
+
+Now the Ui component is ready to take inputs from the User, and the input is processed and executed
+in the Logic component, the execution result is passed back to Ui component and displayed to the User.
+Just before the main program exits, Storage component saves the content in the TaskMap to the data file.
+
+Commons provide utility functions and messages to be used by other components.
+
+The sequence diagram below shows a typical workflow the program.
+
+
+![ComponentsSeq](diagrams/ComponentsSeq.png)
+
+
+In the example above, after the programs starts, User inputs "delete 2173",
+the input is captured by the Ui component, then it calls the processRaw 
+function from the Logic component, to extract useful information from the input. 
+Logic component verified the command is valid and calls the remove function 
+from the TaskMap, then TaskMap carries out the operation, and a message is
+return to the Ui from the Logic component.
+
+The next input provide by the User is "bye", it follows a similar process.
+After the message is shown to the User, the main program calls the
+writeTasksToFile() function in Storage and saves the data.
+
+
+### Ui component
+
+
+![UiComponent](diagrams/UiClass.png)
+
+
+The diagram above is a class diagram of the Ui component.
+Other than the usual InputStream and OutputStream, the Ui component has a
+DisplayDateStructure. This abstract class prepares the content to be 
+printed when the DisplayMode is WEEK or MONTH. The content is generated
+using generateScreen function, which writes the intended content into
+the 2-D array of characters. After that, the array will be shown to the user
+by Ui class via the OutputStream. 
+
+Moreover, WeekStructure and MonthStructure extends the DisplayDateStructure.
+These classes will generate a different size of the 2-D array mentioned
+in the previous paragraph due the difference in number of days displayed.
+
+### Logic component
+
+
+![LogicComponent](diagrams/LogicClass.png)
+
```

Comment:
For the different commands, I think it's better to name the classes as ListCommand, SearchCommand, HelpCommand, etc. rather than just the keyword themselves.

![image](https://user-images.githubusercontent.com/57692185/97387337-bd3d1480-1910-11eb-979f-1ee5d497df9d.png)